packages/@expo/cli/src/start/server/metro/withMetroMultiPlatform.ts

```diff
@@ -116,17 +116,9 @@ export function withWebResolvers(config: ConfigT, projectRoot: string) {
       const result = resolve(
         {
           ...context,
+          mainFields,
           preferNativePlatform: platform !== 'web',
           resolveRequest: undefined,
-
-          // Passing `mainFields` directly won't be considered
-          // we need to extend the `getPackageMainPath` directly to
-          // use platform specific `mainFields`.
-          getPackageMainPath(packageJsonPath) {
```

Comment:
Would we be able to keep both fields in order to support multiple versions or will that cause problems?

Reply:
In theory, yes — `getPackageMainPath` will still be recognised by earlier Metro versions, but is entirely dropped (along with `context.moduleCache`) in Metro 0.75.0 (since Expo CLI is the only public user of this prop, we elected to move fast in this instance).

So yes perhaps that is a good interim state (with appropriate comments!). Will push an update.